### PR TITLE
netdev_tap: make 'wired' property configurable

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

`netdev_tap` is a virtual interface, make it possible to simulate both a wired and a wireless interface.

### Testing procedure
```
$ bin/native/gnrc_networking.elf tap0

Iface  6  HWaddr: 7A:37:FC:7D:1A:AF 
          L2-PDU:1500  MTU:1500  HL:64  RTR  
          RTR_ADV  
          Source address length: 6
--->      Link type: wired
          inet6 addr: fe80::7837:fcff:fe7d:1aaf  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff7d:1aaf
          inet6 group: ff02::1a
          
          Statistics for Layer 2
            RX packets 4  bytes 480
            TX packets 1 (Multicast: 1)  bytes 78
            TX succeeded 1 errors 0
          Statistics for IPv6
            RX packets 4  bytes 424
            TX packets 1 (Multicast: 1)  bytes 64
            TX succeeded 1 errors 0
```

```
$ bin/native/gnrc_networking.elf -w tap0

Iface  6  HWaddr: 7A:37:FC:7D:1A:AF 
          L2-PDU:1500  MTU:1500  HL:64  RTR  
          RTR_ADV  
          Source address length: 6
--->      Link type: wireless
          inet6 addr: fe80::7837:fcff:fe7d:1aaf  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff7d:1aaf
          inet6 group: ff02::1a
          
          Statistics for Layer 2
            RX packets 4  bytes 480
            TX packets 1 (Multicast: 1)  bytes 78
            TX succeeded 1 errors 0
          Statistics for IPv6
            RX packets 4  bytes 424
            TX packets 1 (Multicast: 1)  bytes 64
            TX succeeded 1 errors 0

```

```
$ bin/native/gnrc_networking.elf -w tap0 tap1

Iface  7  HWaddr: 5A:B0:08:67:AD:82 
          L2-PDU:1500  MTU:1500  HL:64  RTR  
          RTR_ADV  
          Source address length: 6
          Link type: wired
          inet6 addr: fe80::58b0:8ff:fe67:ad82  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff67:ad82
          
          Statistics for Layer 2
            RX packets 5  bytes 526
            TX packets 1 (Multicast: 1)  bytes 78
            TX succeeded 1 errors 0
          Statistics for IPv6
            RX packets 5  bytes 456
            TX packets 1 (Multicast: 1)  bytes 64
            TX succeeded 1 errors 0

Iface  6  HWaddr: 7A:37:FC:7D:1A:AF 
          L2-PDU:1500  MTU:1500  HL:64  RTR  
          RTR_ADV  
          Source address length: 6
          Link type: wireless
          inet6 addr: fe80::7837:fcff:fe7d:1aaf  scope: link  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ff7d:1aaf
          
          Statistics for Layer 2
            RX packets 4  bytes 480
            TX packets 1 (Multicast: 1)  bytes 78
            TX succeeded 1 errors 0
          Statistics for IPv6
            RX packets 4  bytes 424
            TX packets 1 (Multicast: 1)  bytes 64
            TX succeeded 1 errors 0
```

#### LWIP

```
$  bin/native/paho-mqtt-example.elf -w tap0 tap1

Iface ET0 HWaddr: 7a:37:fc:7d:1a:af Link: up State: up
        Link type: wireless
        inet6 addr: fe80:0:0:0:7837:fcff:fe7d:1aaf scope: link
Iface ET1 HWaddr: 5a:b0:08:67:ad:82 Link: up State: up
        Link type: wired
        inet6 addr: fe80:0:0:0:58b0:8ff:fe67:ad82 scope: link
```

### Issues/PRs references

https://github.com/RIOT-OS/RIOT/pull/17350#discussion_r814892585

### CODE REVIEW

**Comment by miri64** [comment]
Wouldn't it make more sense to make this compile-time configurable? Also: before we do this, can we consider, if this is actually nececessary? See https://github.com/RIOT-OS/RIOT/pull/17350#discussion_r814978759

**Comment by benpicco** [comment]
I thought maybe you want to have a wired and a wireless interface :man_shrugging: 
But I can add a `netdev_tap_wireless` pseudo-module instead

**Comment by miri64** [comment]
> I thought maybe you want to have a wired and a wireless interface man_shrugging

Right, that makes sense! But you could also do that in `native_tap_params_t`, right?

**Comment by miri64** [comment]
> But I can add a `netdev_tap_wireless` pseudo-module instead

I'd rather go with a Kconfig switch, a pseudo-module seems a bit overkill IMHO.

**Comment by benpicco** [comment]
Let's make it a command-line argument, that also sorts out the issue of which interface should be wired.

File: cpu/native/include/netdev_tap.h [hunk 1]
```diff
@@ -43,7 +44,8 @@ typedef struct netdev_tap {
     char tap_name[IFNAMSIZ];            /**< host dev file name */
     int tap_fd;                         /**< host file descriptor for the TAP */
     uint8_t addr[ETHERNET_ADDR_LEN];    /**< The MAC address of the TAP */
-    uint8_t promiscuous;                 /**< Flag for promiscuous mode */
+    uint8_t promiscuous;                /**< Flag for promiscuous mode */
+    uint8_t wired;                      /**< Flag for wired mode */
```
Comment (miri64) [inline cpu/native/include/netdev_tap.h:48]: Why not use `bool` here? I only recently learned that it is also 8 bit wide :-)

Comment (benpicco) [reply cpu/native/include/netdev_tap.h]: Just consistency.

**Review by miri64 — APPROVED** [review]
ACK. Tested. Might also help with future testing, as now the tap interface becomes basically optional.

**Pull request merged**